### PR TITLE
[Refactor-4856]: make childrenprops explicit

### PR DESCRIPTION
**Pull request by SireeKoolenWijkstra**
Made childrenprops explicit in preparation to upgrade to React 18. In accordance with Oscar's suggestion, where children are implicit the component is made a function and react.fc has been removed.

Ticket: [SIG-4856](https://datapunt.atlassian.net/browse/SIG-4856)

## Signalen

Before opening a pull request, please ensure:

- [ ] Double-check your branch is based on `main` and targets `main`
- [ ] Pull request has tests (we are going for 100% coverage!)
- [ ] Code is well-commented, linted and follows project conventions
- [ ] Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/components/ButtonBar/index.tsx [hunk 1]
```diff
@@ -12,14 +13,14 @@ const Bar = styled.div`
 
 export interface ButtonBarProps {
   className?: string
+  children?: ReactNode
```
Comment (vdegraaf) [inline src/components/ButtonBar/index.tsx:16]: I think children are required here. Else you don't want to render anything.

File: src/components/IconList/IconList.tsx [hunk 2]
```diff
@@ -27,30 +29,33 @@ export interface IconListItemProps {
...
-        alt=""
-        height={20}
-        src={featureStatusType.icon.iconUrl}
-        width={20}
-      />
-    )}
-    {children}
-  </StyledListItem>
-)
+}: IconListItemProps): JSX.Element {
```
Comment (vdegraaf) [inline src/components/IconList/IconList.tsx:42]: You don't have to add the `JSX.Element` here, that is the implicit return type already. Also can leave the implicit return.

File: src/components/ThemeProvider/index.tsx [hunk 3]
```diff
@@ -54,14 +55,19 @@ export const getConfig: (theme?: RecursivePartial<Theme>) => Theme = (
...
-  >
-    {children}
-  </ASCThemeProvider>
-)
+interface ThemeProviderProps {
+  children: ReactNode
+}
 
+function ThemeProvider({ children }: ThemeProviderProps): JSX.Element {
+  return (
```
Comment (vdegraaf) [inline src/components/ThemeProvider/index.tsx:63]: See previous comments.

File: src/signals/IncidentMap/components/DrawerOverlay/DrawerOverlay.tsx [hunk 4]
```diff
@@ -28,15 +28,16 @@ export interface Props {
...
-export const DrawerOverlay: FunctionComponent<Props> = ({
+export function DrawerOverlay({
   children,
   incident,
   onCloseDetailPanel,
   onStateChange,
   state = DrawerState.Closed,
-}) => {
+}: Props): JSX.Element {
   const mode = useDeviceMode()
```
Comment (vdegraaf) [inline src/signals/IncidentMap/components/DrawerOverlay/DrawerOverlay.tsx:41]: Same

**Review by vdegraaf — COMMENTED** [review]
Hola!
I think you made some unnecessary changes. 
- The FC type etc need to be removed. That's good.
- add the proptypes like  {}: Props. Also when good.
- But then you removed the arrow functions and added explicit return. Which is not necessary and different then we do elsewhere in the code. 
- Also, the JSX.Element (or null) is the implicit return type of a react component. So you can leave that out.

**Comment by OscarBakker** [comment]
Functions are fine but for consistency it's a bit better to use arrows instead. Also a bit denser. Not wrong, just arrows are preferable.

File: src/components/ButtonBar/index.tsx [hunk 5]
```diff
@@ -12,14 +13,14 @@ const Bar = styled.div`
...
 
 /**
  * Button container that merely adds margin between children
  */
-const ButtonBar: FunctionComponent<ButtonBarProps> = ({
-  className,
-  children,
-}) => <Bar className={className}>{children}</Bar>
+function ButtonBar({ className, children }: ButtonBarProps): JSX.Element {
+  return <Bar className={className}>{children}</Bar>
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/ButtonBar/index.tsx:23]: Done

File: src/components/Markdown/Markdown.tsx [hunk 6]
```diff
@@ -1,39 +1,47 @@
...
-  >
-    {children}
-  </ReactMarkdown>
-)
+function Markdown({
+  children,
+  hideTabindexLink,
+  ...props
+}: Props): JSX.Element {
+  return (
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/Markdown/Markdown.tsx:26]: Done

File: src/components/PreviousButton/PreviousButton.tsx [hunk 7]
```diff
@@ -21,21 +23,23 @@ interface PreviousButtonProps {
...
-    iconSize={14}
-    onClick={onClick}
-    type="button"
-    variant="textButton"
-  >
-    {children}
-  </StyledButton>
-)
+}: PreviousButtonProps): JSX.Element {
+  return (
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/PreviousButton/PreviousButton.tsx:31]: done

File: src/components/ThemeProvider/index.tsx (hunk 3 above)
Comment (SireeKoolenWijkstra) [reply src/components/ThemeProvider/index.tsx:63]: Done

File: src/components/pages/BasePage/BasePage.tsx [hunk 8]
```diff
@@ -30,32 +32,34 @@ const ContentWrapper = styled.div`
...
-        )}
-
-        {children}
-      </article>
-    </ContentWrapper>
-  </Row>
-)
+}: BasePageProps): JSX.Element {
+  return (
+    <Row data-testid="basePage" {...props}>
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/pages/BasePage/BasePage.tsx:42]: done

File: src/signals/incident-management/components/Status/Status.tsx [hunk 9]
```diff
@@ -21,8 +22,8 @@ const StyledStatus = styled.span<{ statusCode?: StatusCode; theme: Theme }>`
       : theme.colors.support.invalid};
 `
 
-const Status: FC<StatusProps> = ({ children, statusCode }) => (
-  <StyledStatus statusCode={statusCode}>{children}</StyledStatus>
-)
+function Status({ children, statusCode }: StatusProps): JSX.Element {
+  return <StyledStatus statusCode={statusCode}>{children}</StyledStatus>
+}
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/components/Status/Status.tsx:27]: done

File: src/signals/incident-management/containers/IncidentDetail/components/FileInput/FileInput.tsx [hunk 10]
```diff
@@ -8,16 +11,17 @@ interface FileInputProps {
...
-const FileInput: FunctionComponent<FileInputProps> = ({
+function FileInput({
   allowedFileTypes = ['image/jpeg', 'image/jpg', 'image/png', 'image/gif'],
   files = [],
   multiple = true,
   name,
   onChange,
   children,
-}) => {
+}: FileInputProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/IncidentDetail/components/FileInput/FileInput.tsx:24]: done

File: src/signals/incident/components/form/FormField/FormField.tsx [hunk 11]
```diff
@@ -59,17 +59,18 @@ export interface FormFieldProps extends Pick<ReactiveFormMeta, PickedProps> {
   meta: FormMeta
   options?: FormOptions
   isFieldSet?: boolean
+  children?: ReactNode
```
Comment (OscarBakker) [inline src/signals/incident/components/form/FormField/FormField.tsx:62]: this should not be optional, a form field always is a parent to some other element right?

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**